PIL/Image.py

```diff
@@ -600,6 +636,7 @@ def _dump(self, file=None, format=None, **options):
         return file
 
     def __eq__(self, other):
+        # type: (object) -> bool
         return (self.__class__.__name__ == other.__class__.__name__ and
                 self.mode == other.mode and
```

Comment:
Right. I know that mypy can use `isinstance` to distinguish in cases like this. Since all the variables that are being tested are in `Image`, would adding `isinstance(other, Image)` before or after the first test be reasonable from a code sense? It does resolve the typing issue, at least here.